usrp3/tools/scripts/uhd_image_builder.py

```diff
@@ -133,20 +138,87 @@ def setup_parser():
     )
     return parser
 
-def create_vfiles(args):
+def get_default_parameters():
+    default = {"clock" : "ce",
+               "parameters" : None,
+               "extraports" : None}
+    return default
+
+
+def parse_yml(ymlfile):
+    """
+    Parse an input yaml file with a list of blocks and parameters!
+    """
+    try:
+        import yaml
+    except ImportError:
+        print('[ERROR] Could not import yaml module')
+        exit(1)
+
+    with open(ymlfile, 'r') as f:
+        data = yaml.load(f)
+    blocks = []
+    params = []
+    for val in data:
+        print(val['block'])
+        blocks.append(val['block'])
+        blockparams = get_default_parameters()
+        if "clock" in val:
+            blockparams["clock"] = val["clock"]
+        if "parameters" in val:
+            blockparams["parameters"] = val["parameters"]
+        if "extraports" in val:
+            blockparams["extraports"] = val["extraports"]
+        print(blockparams)
+        params.append(blockparams)
+    print(data)
+    return blocks, params
+
+def format_param_str(parameters):
+    """
+    Take a single block parameter dictionary and format as a verilog string
+    """
+    paramstr = ""
+    if parameters:
+        paramstrlist = []
+        for key in parameters.keys():
+            value = ""
+            if not (parameters[key] is None):
+                value = parameters[key]
+            currstr = ".%s(%s)" % (str.upper(key), value)
+            paramstrlist.append(currstr)
+        paramstr = "#(%s)" % (", ".join(paramstrlist))
+    return paramstr
+
+def format_port_str(extraports):
+    """
+    Take a single dictionary and format as a verilog string representing extra block ports
+    """
+    portstr = ""
+    if extraports:
+        portstrlist = []
+        for key in extraports.keys():
+            value = ""
+            if not (extraports[key] is None):
```

Comment:
PyLint will probably complain about this formatting.

Reply:
Why's that? I don't usually use pylint. From the double negative?

Reply:
I strongly recommend using it. You can run it from many editors, e.g. vim, via plugins. It's the closest we get to a `-Wall` for Python. In this case, it's non-standard use of parentheses and would suggest you rewrite it as `if extraports[key] is not None`.